clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/DestroyTitusJob.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.titus.deploy.actions;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.netflix.spinnaker.clouddriver.event.EventMetadata;
+import com.netflix.spinnaker.clouddriver.saga.SagaCommand;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.models.Saga;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.titus.TitusClientProvider;
+import com.netflix.spinnaker.clouddriver.titus.client.TitusClient;
+import com.netflix.spinnaker.clouddriver.titus.client.model.Job;
+import com.netflix.spinnaker.clouddriver.titus.client.model.TerminateJobRequest;
+import com.netflix.spinnaker.clouddriver.titus.credentials.NetflixTitusCredentials;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.DestroyTitusJobDescription;
+import javax.annotation.Nonnull;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DestroyTitusJob implements SagaAction<DestroyTitusJob.DestroyTitusJobCommand> {
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final TitusClientProvider titusClientProvider;
+
+  @Autowired
+  public DestroyTitusJob(
+      AccountCredentialsProvider accountCredentialsProvider,
+      TitusClientProvider titusClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.titusClientProvider = titusClientProvider;
+  }
+
+  @NotNull
+  @Override
+  public Result apply(@NotNull DestroyTitusJob.DestroyTitusJobCommand command, @NotNull Saga saga) {
+    saga.log(
+        "Destroying Titus Job %s:%s:%s",
+        command.description.getAccount(),
+        command.description.getRegion(),
+        command.description.getJobId());
+
+    AccountCredentials accountCredentials =
+        accountCredentialsProvider.getCredentials(command.description.getAccount());
+
+    TitusClient titusClient =
+        titusClientProvider.getTitusClient(
+            (NetflixTitusCredentials) accountCredentials, command.description.getRegion());
+
+    Job job = titusClient.getJobAndAllRunningAndCompletedTasks(command.description.getJobId());
```

Comment:
This call could fail. Could handle the exception and make it retryable?

Reply:
Whole point of the saga right? The `TitusExceptionHandler` is going to flag the action as retryable based on its criteria and `orca` will automatically retry.

Reply:
Yup, didn’t notice the exception handler at first glance.